src/Sylius/Bundle/ChannelBundle/Context/FakeChannel/FakeChannelCollector.php

```diff
@@ -30,7 +31,12 @@
     public function __construct(ChannelRepositoryInterface $channelRepository, ChannelContextInterface $channelContext)
     {
         $this->data['channels'] = $channelRepository->findAll();
-        $this->data['current_channel'] = $channelContext->getChannel();
+
+        try {
+            $this->data['current_channel'] = $channelContext->getChannel();
+        } catch (ChannelNotFoundException $exception) {
+            $this->data['current_channel'] = null;
```

Comment:
Other option instead of using `default()` filter in Twig, would be to set "fake" channel here, WDYT?